flows/actions/modifiers/status.go

```diff
@@ -0,0 +1,55 @@
+package modifiers
+
+import (
+	"encoding/json"
+
+	"github.com/nyaruka/goflow/assets"
+	"github.com/nyaruka/goflow/envs"
+	"github.com/nyaruka/goflow/flows"
+	"github.com/nyaruka/goflow/flows/events"
+	"github.com/nyaruka/goflow/utils"
+)
+
+func init() {
+	registerType(TypeStatus, readStatusModifier)
+}
+
+// TypeStatus is the type of our status modifier
+const TypeStatus string = "status"
+
+// StatusModifier modifies the status of a contact
+type StatusModifier struct {
+	baseModifier
+
+	Status flows.ContactStatus `json:"status" validate:"eq=active|eq=blocked|eq=stopped"`
+}
+
+// NewStatus creates a new status modifier
+func NewStatus(status flows.ContactStatus) *StatusModifier {
+	return &StatusModifier{
+		baseModifier: newBaseModifier(TypeStatus),
+		Status:       status,
+	}
+}
+
+// Apply applies this modification to the given contact
+func (m *StatusModifier) Apply(env envs.Environment, assets flows.SessionAssets, contact *flows.Contact, log flows.EventCallback) {
+
+	changed, blockedOrStopped := contact.SetStatus(m.Status)
```

Comment:
This can be implemented just like the other modifiers that operate on one field - just compare `contact.Status()` with `m.Status` here

You also don't need `blockedOrStopped`.. you're passing contact to `m. reevaluateGroups` so it can look at the status there